src/Command/PluginCommand.php

```diff
@@ -91,34 +97,36 @@ public function main($name = null)
      * Also update the autoloader and the root composer.json file if it can be found
      *
      * @param string $plugin Name of the plugin in CamelCased format
+     * @param \Cake\Console\Arguments $args The command arguments.
+     * @param \Cake\Console\ConsoleIo $io The console io
      * @return bool|null
      */
-    public function bake($plugin)
+    public function bake($plugin, $args, $io)
```

Comment:
why we could not use  `$this->io` object with assigning these vars withing `execute()` ?

Reply:
We could. I've been passing the arguments around in other commands so that the commands have as few properties as possible. I'm also considering making only `execute()` public and all other methods protected.